pkg/network/apis/network/validation/validation.go

```diff
@@ -24,37 +26,42 @@ func SetDefaultClusterNetwork(cn networkapi.ClusterNetwork) {
 // ValidateClusterNetwork tests if required fields in the ClusterNetwork are set, and ensures that the "default" ClusterNetwork can only be set to the correct values
 func ValidateClusterNetwork(clusterNet *networkapi.ClusterNetwork) field.ErrorList {
 	allErrs := validation.ValidateObjectMeta(&clusterNet.ObjectMeta, false, path.ValidatePathSegmentName, field.NewPath("metadata"))
+	var testedCIDRS []*net.IPNet
 
-	clusterIPNet, err := netutils.ParseCIDRMask(clusterNet.Network)
```

Comment:
Should validate that either Network and HostSubnetLength are unset, or that they are identical to the first element of ClusterNetworks.

Reply:
That happens in the master config validation/conversion. If both ClusterNetworks/Hostsubnet are set and ClusterNetworks are set Clayton wants startup to fail and report that as an error

Reply:
That's the config object. This is the API object. They're separate.

Although what I said here will do the wrong thing for user-created non-"default" ClusterNetwork objects. So... ok, we should validate that if Network and/or HostSubnetLength is set, then they are valid values, but we don't need to compare them against ClusterNetworks.

Oh, except that if ClusterNetworks is 0-length, then Network/HostSubnetLength need to be set.

Reply:
I never set the api object for network or hostsubnetlengh, in pkg/cmd/server/api/v1/conversions.go I use DeprecatedHostSubnetLength and DeprecatedClusterNetwork to build a clusterNetworkEntry if len(clusterNetworkEntry) = 0 and unset the deprecated ones.